### PR TITLE
HeapSort with Readme

### PR DESCRIPTION
**Pull request by nitishtw**

### CODE REVIEW

File: sort/BucketSort.cpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+// Generally used for floating point numbers.
+
+
+void bucketSort(float a[],int n){
+vector<float> buckets[n];
+int i,j;
+for(i=0;i<n;i++){
+    buckets[(int)(n*a[i])].push_back(a[i]);
+}
+for(i=0;i<n;i++){
```
Comment (prateekiiest) [inline sort/BucketSort.cpp]: document the code for understanding

File: sort/BucketSort.cpp [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+*/
+
+// Generally used for floating point numbers.
+
+
+void bucketSort(float a[],int n){
+vector<float> buckets[n];
+int i,j;
+for(i=0;i<n;i++){
+    buckets[(int)(n*a[i])].push_back(a[i]);
```
Comment (prateekiiest) [inline sort/BucketSort.cpp]: I didn't get this line. can you elaborate?

File: sort/HeapSort.cpp [hunk 3]
```diff
@@ -0,0 +1,52 @@
+#include<iostream>
+using namespace std;
+
+
+//-------------------------------------------Heapify
+void maxHeapify(int *a,int i,int n){
+//Runs in O(logN) , floats down value at index i until a[i] becomes a max heap
```
Comment (prateekiiest) [inline sort/HeapSort.cpp]: dont write complexities here, write in a separate readme

File: sort/RadixSort.cpp [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    for(i=0;i<n;i++)
+        a[i]=output[i];
+
+}
+
+//------------------------------------------Radix Sort
+void radixSort(int *a,int n){
+
+int m = getMax(a,n);
+for(int exp=1;m/exp > 0; exp=exp*10){
```
Comment (prateekiiest) [inline sort/RadixSort.cpp]: could you pls explain why exponential is being used in this case?

File: sort/RadixSort.cpp [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    return mx;
+}
+//-------------------------------------Counting Sort SubRoutine
+void countingSort(int *a,int n,int exp){
+    int output[n];
+    int count[10];
+    memset(count,0,sizeof count);
+    int i;
+
+    for(i=0;i<n;i++){
```
Comment (prateekiiest) [inline sort/RadixSort.cpp]: document code pls

**Review by prateekiiest — CHANGES_REQUESTED** [review]

**Comment by prateekiiest** [comment]
please make sure you make a separate folder for all the code. Like for heapsort make a folder named heapsort and there you add the code and a readme for the analysis.
Same for radix, bucket.

And see the review

**Comment by nitishtw** [comment]
Ok sir,  i have got it... I'll fix it soon!

File: sort/HeapSort/Readme [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+4 largest = l
+5 else largest = i
+6 if r <= A.heap-size and A[r] > A[largest]
+7 largest = r
+8 if largest != i
+9 exchange A[i] with A[largest]
+10 MAX-HEAPIFY(A,largest)
+
+BUILD-MAX-HEAP(A)
+1 A.heap-size = A.length
```
Comment (prateekiiest) [inline sort/HeapSort/Readme]: dont include the algo pseudo code here

**Review by prateekiiest — COMMENTED** [review]
dont include the algo pseudo code here

**Comment by nitishtw** [comment]
ohk sir, I hav removed it.
Now plz have a look on it once again

**Comment by nitishtw** [comment]
If this format is ok then I'll upload further in this format only...Thus
please check it once !!

File: sort/HeapSort/HeapSort.cpp [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+//----------------------------Print Function
+void print(int *a,int n){
+    for(int i=0;i<5;i++)
+        cout<<a[i]<<" "<<endl;
+
+}
+
+int main(){
+
+int a[]={5,1,2,0,6};
```
Comment (abiduzz420) [inline sort/HeapSort/HeapSort.cpp]: It would be good if you can dynamically allocate the array @AdvikEshan

**Comment by nitishtw** [comment]
I'll change it by tomorrow... Thanx!

**Comment by nitishtw** [comment]
Sir, Change has been made to HeapSort (Array is allocated dynamically) !! Have a look !

**Comment by abiduzz420** [comment]
@AdvikEshan You are still declaring array as `int a[20]`. That's not dynamic memory allocation

**Comment by nitishtw** [comment]
Oh Sorry, i have got the wrong meaning ! My mistake !! I'll change it soon.

**Comment by nitishtw** [comment]
Now, I have updated it...Have a look @abiduzz420 @prateekiiest  !! #68

**Comment by soubh1k** [comment]
I am closing this pull request. The current directory structure has been changed. Fork the new repo with the new directory structure. Insert the files in the new directory structure along with the appropriate readme's. Then create a new Pull Request.

**Pull request closed without merging**